api/blockindexer/indexer.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txindexer
+
+import (
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/event"
+	"github.com/ava-labs/hypersdk/internal/cache"
+	"github.com/ava-labs/hypersdk/vm"
+)
+
+const (
+	Name      = "blockindexer"
+	Namespace = "blockindexer"
+)
+
+var (
+	_ event.SubscriptionFactory[*chain.StatefulBlock] = (*subscriptionFactory)(nil)
+	_ event.Subscription[*chain.StatefulBlock]        = (*indexer)(nil)
+)
+
+type Config struct {
+	Enabled             bool `json:"enabled"`
+	AcceptedBlockWindow int  `json:"acceptedBlockWindow"`
+}
+
+func NewDefaultConfig() Config {
+	return Config{
+		Enabled:             true,
+		AcceptedBlockWindow: 1_000,
+	}
+}
+
+func With() vm.Option {
```

Comment:
I don't know when we adopted this convention but I think `With` reads weirdly - i think `blockindexer.New()` reads a lot better.

Reply:
Just following the existing style. I think `With` is just an options convention, but open to switching all of them in a separate PR.